packages/viewer/src/modules/Viewer.ts

```diff
@@ -107,7 +108,7 @@ export class Viewer extends EventEmitter implements IViewer {
     return extension as T
   }
 
-  public getExtension<T extends Extension>(type: typeof Extension): T {
+  public getExtension<T extends Extension>(type: Constructor<T>): T {
```

Comment:
@AlexandruPopovici  The way this function was typed was off, which is why you had to manually add the <CameraController> generic argument everywhere (`getExtension<CameraController>(CameraController)`). There were usages in FE2 that did not have that generic argument so obviously those were still broken.

I fixed the function type to correctly resolve the class type from the generic argument (previously you had `typeof Extension` which was in no way related to the generic T arg, so no inference was made), so now it should work even without having to manually specify the generic everywhere.

You should go through all of those usages where you added the arg and remove them